store/store.go

```diff
@@ -437,24 +486,26 @@ func (s *Store) doRequest(client *http.Client, reqOptions *requestOptions, user
 
 	wwwAuth := resp.Header.Get("WWW-Authenticate")
 	if resp.StatusCode == 401 && user != nil && strings.Contains(wwwAuth, "needs_refresh=1") {
-		// close previous response, refresh and retry
+		// close previous response, refresh user and retry
 		resp.Body.Close()
-		err = refreshMacaroon(user)
+		err = s.refreshUser(user)
 		if err != nil {
 			return nil, err
 		}
-		if s.authContext != nil {
-			err = s.authContext.UpdateUser(user)
-			if err != nil {
-				return nil, err
-			}
-		}
-		req, err := s.newRequest(reqOptions, user)
+		// retry
+		return s.doRequest(client, reqOptions, user)
+
+	} else if resp.StatusCode == 401 && strings.Contains(wwwAuth, "refresh_device_session=1") {
```

Comment:
@matiasb no, my question is can we get back both needs_refresh=1 and refresh_device_session=1 in one response? should we have code to do both at the same time then?

I'm asking to always refresh both